src/pkg/path/category_type.go

```diff
@@ -29,31 +29,47 @@ const (
 	ChannelMessagesCategory CategoryType = 9 // channelMessages
 )
 
-func ToCategoryType(category string) CategoryType {
-	cat := strings.ToLower(category)
-
-	switch cat {
-	case strings.ToLower(EmailCategory.String()):
-		return EmailCategory
-	case strings.ToLower(ContactsCategory.String()):
-		return ContactsCategory
-	case strings.ToLower(EventsCategory.String()):
-		return EventsCategory
-	case strings.ToLower(FilesCategory.String()):
-		return FilesCategory
-	case strings.ToLower(LibrariesCategory.String()):
-		return LibrariesCategory
-	case strings.ToLower(ListsCategory.String()):
-		return ListsCategory
-	case strings.ToLower(PagesCategory.String()):
-		return PagesCategory
-	case strings.ToLower(DetailsCategory.String()):
-		return DetailsCategory
-	case strings.ToLower(ChannelMessagesCategory.String()):
-		return ChannelMessagesCategory
-	default:
-		return UnknownCategory
+var strToCat = map[string]CategoryType{
+	strings.ToLower(EmailCategory.String()):           EmailCategory,
+	strings.ToLower(ContactsCategory.String()):        ContactsCategory,
+	strings.ToLower(EventsCategory.String()):          EventsCategory,
+	strings.ToLower(FilesCategory.String()):           FilesCategory,
+	strings.ToLower(LibrariesCategory.String()):       LibrariesCategory,
+	strings.ToLower(ListsCategory.String()):           ListsCategory,
+	strings.ToLower(PagesCategory.String()):           PagesCategory,
+	strings.ToLower(DetailsCategory.String()):         DetailsCategory,
+	strings.ToLower(ChannelMessagesCategory.String()): ChannelMessagesCategory,
+}
+
+func ToCategoryType(s string) CategoryType {
+	cat, ok := strToCat[strings.ToLower(s)]
+	if ok {
+		return cat
 	}
+
+	return UnknownCategory
+}
+
+var catToHuman = map[CategoryType]string{
+	EmailCategory:           "Emails",
+	ContactsCategory:        "Contacts",
+	EventsCategory:          "Events",
+	FilesCategory:           "Files",
+	LibrariesCategory:       "Libraries",
+	ListsCategory:           "Lists",
+	PagesCategory:           "Pages",
+	DetailsCategory:         "Details",
+	ChannelMessagesCategory: "Messages",
+}
+
+// HumanString produces a more human-readable string version of the category.
+func (cat CategoryType) HumanString() string {
```

Comment:
this is starting to feel more like we should be defining a struct for it rather than creating more and more maps

Reply:
I've tossed around the same idea in my head for a while.  Ultimately, unless we expand the behavior even further (I don't think that feels very likely, personally) the lift outweighs the value at this moment.  It's a two-way door, and always an option for the future.